### PR TITLE
[RHICOMPL-888] README: Document architecture and update dev setup

### PR DESCRIPTION
**Pull request by vkrizan**
Adds list of dependent services and a list of components. Updates dev setup.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,12 +9,44 @@ a policy. For example, you should be able to generate reports of all kinds for
...
 
 
-## Getting started
+## Architecture
 
 This project does two main things:
 
-1 - Connect to a Kafka message queue [provided by the Insights Platform](https://github.com/RedHatInsights/insights-upload)
-2 - Serve as the API backend for the web UI [compliance-frontend](https://github.com/RedHatInsights/compliance-frontend) and for other consumers.
+1. Serve as the API/GraphQL backend for the web UI
```
Comment (akofink) [inline README.md:16]: GraphQL is an API. If you want to differentiate them, we can mention REST vs. GraphQL.

File: README.md [hunk 2]
```diff
@@ -9,12 +9,44 @@ a policy. For example, you should be able to generate reports of all kinds for
...
+   [compliance-frontend](https://github.com/RedHatInsights/compliance-frontend)
+   and for other consumers,
+2. Connect to a Kafka message queue provided by the Insights Platform.
+
+### Components
+
+The Insights Compliance backend comprises of these components/services:
+
+* Rails web server — serving API and GrapQL (port 3000)
```
Comment (akofink) [inline README.md:25]:
```suggestion
* Rails web server — serving API and GraphQL (port 3000)
```

File: README.md [hunk 3]
```diff
@@ -9,12 +9,44 @@ a policy. For example, you should be able to generate reports of all kinds for
...
+* Sidekiq — job runner connected through Redis
```
Comment (akofink) [inline README.md:26]:
```suggestion
* Sidekiq — job runner connected through Redis (see app/jobs)
```

**Review by akofink — COMMENTED** [review]
Looks great! Just a few minor comments.

File: README.md (hunk 1 above)
Comment (vkrizan) [reply README.md:16]: Differentiated below.

**Review by akofink — APPROVED** [review]
Thanks, @vkrizan!

**Pull request merged**